src/typography/src/Heading.js

```diff
@@ -3,27 +3,35 @@ import cx from 'classnames'
 import PropTypes from 'prop-types'
 import Box from 'ui-box'
 import { useStyleConfig } from '../../hooks'
+import { useTheme } from '../../theme'
 
 const pseudoSelectors = {}
 const internalStyles = {}
 
 const Heading = memo(
   forwardRef(function Heading(props, ref) {
-    const { className, size = 500, ...restProps } = props
+    const { className, color: colorProp = 'default', size = 500, ...restProps } = props
     const { className: themedClassName, ...styleProps } = useStyleConfig(
       'Heading',
       { size },
       pseudoSelectors,
       internalStyles
     )
 
+    const { colors } = useTheme()
+
+    const color = colorProp === 'none' || colorProp === 'default' ? 'default' : colorProp
+
+    const themedColor = colors[color] || (colors.text && colors.text[color]) || color
+
     return (
       <Box
         is="h2"
         ref={ref}
         className={cx(themedClassName, className)}
         marginTop={0}
         marginBottom={0}
+        color={themedColor}
```

Comment:
This seems reasonable to me, but note that it will get clobbered by `color` defined in the theme hook for this particular component. That means you can't do instance-specific overrides for color if the component's theme specifies its own.

Reply:
That's true, it's probably also not a good idea to have the heading rely on the theme hook. The original intent was to make it more convenient but if this introduces unexpected behaviour then it's perhaps best not to include this PR.  Or do you perhaps have an alternative approach?

Reply:
I would opt for fixing this in the component-specific theme instead, I think. You can pass `color` to the `useStyleConfig` hook.